R/get_source_expressions.R

```diff
@@ -152,6 +152,30 @@ lint_parse_error <- function(e, source_expression) {
   lint_parse_error_nonstandard(e, source_expression)
 }
 
+#' Ensure a string is valid for printing
+#'
+#' Helper to ensure a valid string is provided as line where necessary.
+#' Handles two cases:
+#'
+#'  1. NA lines, as generated by running outside of code blocks in Rmd documents
+#'  2. invalid string lines, as generated by invalid encoding settings
+#'
+#' [nchar()] can detect both cases: 1. returns `NA_integer_`, 2. errors
+#'
+#' @param line Possibly misencoded or NA line
+#'
+#' @return `line` if it's a valid non-NA string, otherwise an empty string.
+#'
+#' @noRd
+fixup_line <- function(line) {
+  nchars <- tryCatch(nchar(line, type = "chars"), error = function(e) NA_integer_)
```

Comment:
tempting to use `NA` here, since we're only checking `is.na()`, since technically `logical` type _could_ be smaller than integer... OTOH maybe `NULL` would allocate even less space. OTOOH this doesn't really matter. Moving on.

Reply:
``` r
pryr::object_size(NA)
#> 56 B
pryr::object_size(NA_integer_)
#> 56 B
```

<sup>Created on 2022-12-08 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

I'd prefer type safety (`nchars` is always an integer) instead of code brevity here.